Content.Shared/VendingMachines/Components/VendingMachineEjectComponent.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Robust.Shared.Audio;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.VendingMachines.Components;
+
+[RegisterComponent, NetworkedComponent]
+public sealed class VendingMachineEjectComponent: Component
+{
+    /// <summary>
+    /// Used by the server to determine how long the vending machine stays in the "Deny" state.
+    /// Used by the client to determine how long the deny animation should be played.
+    /// </summary>
+    [DataField("denyDelay")]
+    public float DenyDelay = 2.0f;
+
+    /// <summary>
+    /// Used by the server to determine how long the vending machine stays in the "Eject" state.
+    /// The selected item is dispensed afer this delay.
+    /// Used by the client to determine how long the deny animation should be played.
+    /// </summary>
+    [DataField("delay")]
+    public float Delay = 1.2f;
+
+    /// <summary>
+    /// When true, will forcefully throw any object it dispenses
+    /// </summary>
+    [DataField("speedLimiter")]
+    public bool CanShoot = false;
+
+    /// <summary>
+    ///     Sound that plays when ejecting an item
+    /// </summary>
+    [DataField("soundVend")]
+    // Grabbed from: https://github.com/discordia-space/CEV-Eris/blob/f702afa271136d093ddeb415423240a2ceb212f0/sound/machines/vending_drop.ogg
+    public SoundSpecifier SoundVend = new SoundPathSpecifier("/Audio/Machines/machine_vend.ogg")
+    {
+        Params = new AudioParams
+        {
+            Volume = -2f
+        }
+    };
+
+    /// <summary>
+    ///     Sound that plays when an item can't be ejected
+    /// </summary>
+    [DataField("soundDeny")]
+    // Yoinked from: https://github.com/discordia-space/CEV-Eris/blob/35bbad6764b14e15c03a816e3e89aa1751660ba9/sound/machines/Custom_deny.ogg
+    public SoundSpecifier SoundDeny = new SoundPathSpecifier("/Audio/Machines/custom_deny.ogg");
+
+    public bool Ejecting;
+    public bool Denying;
+
+    public string? NextItemToEject;
+    public bool ThrowNextItem = false;
+
+    public float NonLimitedEjectForce = 7.5f;
+    public float NonLimitedEjectRange = 5f;
+    public float EjectAccumulator = 0f;
+    public float DenyAccumulator = 0f;
```

Comment:
Replace the accumulators with the time that it will leave the eject/deny state. Use `TimeOffsetSerializer`.